### PR TITLE
Disable downloading files in Bucket and Packages tabs separately

### PR DESCRIPTION
**Pull request by fiskus**
# Description

Added `downloadObject` for disabling download files and directories under the "Bucket" tab. And added `downloadPackaged` for disabling download files and directories in packages.

# TODO

- [x] Unit tests
- [x] Documentation
- [x] [Changelog](../tree/master/docs/CHANGELOG.md) entry (skip if change is not significant to end users, e.g. docs only)

### CODE REVIEW

**Review by nl0 — COMMENTED** [review]
"downloadPackage**d**" feels somewhat awkward. why not just "downloadPackage"?

**Comment by fiskus** [comment]
We allow to download files that are packaged with this option. For `downloadPackage` I would expect the option to download entire package.

**Comment by fiskus** [comment]
Could be also `downloadRawFiles` and `downloadFilesInPackages`, for example

**Comment by nl0** [comment]
> We allow to download files that are packaged with this option. For `downloadPackage` I would expect the option to download entire package.

looking at the code, i can see it affects downloading _anything_ related to packages: whole packages, sub-packages (aka directories), individual package entries

**Comment by nl0** [comment]
also, pls use the changelog in the catalog dir

File: catalog/CHANGELOG.md [hunk 1]
```diff
@@ -16,4 +16,4 @@ where verb is one of
 
 ## Changes
 
-- [Added] Bootstrap the change log ([#XXXX](https://github.com/quiltdata/quilt/pull/XXXX))
```
Comment (nl0) [inline catalog/CHANGELOG.md:19]: oh shi~ i forgot to insert the actual PR number 🤦 
but let's keep that entry anyways (might as well fix it)
could you please bring it back and set the PR number to 4112?

File: catalog/CHANGELOG.md [hunk 2]
```diff
@@ -16,4 +16,4 @@ where verb is one of
...
+* [Added] Add `ui.actions.downloadObject` and `ui.actions.downloadPackage` for configuring visibility of buttons for downloading files and directories under "Bucket" and "Packages" respectively ([#4111](https://github.com/quiltdata/quilt/pull/4111))
```
Comment (nl0) [inline catalog/CHANGELOG.md:19]:
```suggestion
- [Added] Support `ui.actions.downloadObject` and `ui.actions.downloadPackage` options for configuring visibility of download buttons under "Bucket" and "Packages" respectively ([#4111](https://github.com/quiltdata/quilt/pull/4111))
```

File: docs/Catalog/Preferences.md [hunk 3]
```diff
@@ -53,6 +55,10 @@ ui:
 * `ui.actions.createPackage: False` - hide buttons to create packages via
 drag-and-drop or from folders in S3
 * `ui.actions.deleteRevision: True` - show buttons to delete package revision
+* `ui.actions.downloadObject: False` - hide buttons to download
+files and directories under "Bucket" tab
+* `ui.actions.downloadPackage: False` - hide buttons to download
+files and directories under "Packages" tab
```
Comment (nl0) [inline docs/Catalog/Preferences.md:61]: wouldn't this be sufficient?
```suggestion
* `ui.actions.downloadObject: False` - hide download buttons under "Bucket" tab
* `ui.actions.downloadPackage: False` - hide download buttons under "Packages" tab
```

Comment (fiskus) [reply docs/Catalog/Preferences.md:61]: Probably, yes, but I wanted to be specific and express actual intention.

File: catalog/CHANGELOG.md [hunk 4]
```diff
@@ -16,4 +16,5 @@ where verb is one of
 
 ## Changes
 
-- [Added] Bootstrap the change log ([#XXXX](https://github.com/quiltdata/quilt/pull/XXXX))
+- [Added] Bootstrap the change log ([#4112](https://github.com/quiltdata/quilt/pull/4112))
+- [Added] Support `ui.actions.downloadObject` and `ui.actions.downloadPackage` options for configuring visibility of download buttons under "Bucket" and "Packages" respectively ([#4111](https://github.com/quiltdata/quilt/pull/4111))
```
Comment (nl0) [inline catalog/CHANGELOG.md:20]: > Changes are listed in reverse chronological order (newer entries at the top).

```suggestion
- [Added] Support `ui.actions.downloadObject` and `ui.actions.downloadPackage` options for configuring visibility of download buttons under "Bucket" and "Packages" respectively ([#4111](https://github.com/quiltdata/quilt/pull/4111))
- [Added] Bootstrap the change log ([#4112](https://github.com/quiltdata/quilt/pull/4112))
```

**Review by nl0 — APPROVED** [review]

**Pull request merged**